From cdc96e660399b954627e901ed1c8f7618c81a863 Mon Sep 17 00:00:00 2001
From: Vicovan Ionut-Valentin-VCVV001 <Ionut.Vicovan@nxp.com>
Date: Tue, 19 Sep 2017 16:04:31 +0300
Subject: [PATCH 2/2] dtb: add shared mem block as reserved memory node

We add a reserved memory node in the dtb for ls208xa chipsets
to be used as buffers for dma over pcie in bluebox setups.
This is intended to be a temporary solution until a dynamic
reservation mechanism with handshake is implemented.

Signed-off-by: Vicovan Ionut-Valentin-VCVV001 <Ionut.Vicovan@nxp.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
index c19e674..2366757 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
@@ -751,4 +751,19 @@
 		compatible = "fsl,ls-reset";
 		reg = <0x0 0x1E60000 0x0 0x10000>;
 	};
+
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		/* 4Mb shared memory, somewhere at the end of DDRAM */
+		shared_memory: shmem@0x83A0000000 {
+			compatible = "fsl,qoriq-shm";
+			reg = <0x83 0xA0000000 0x0 0x400000>; /* 4 MB */
+			no-map;
+		};
+
+	};
+
 };
-- 
2.7.4

